File: 백준/Silver/2559. 수열/수열.cc
#include<bits/stdc++.h>
using namespace std;

int N, K; // N 개수 , K 연속 
int temp, psum[100001], ret = -10000001;

int main(){
    cin >> N >> K;
    for(int i = 1; i <= N; i++){
        cin >> temp; psum[i] = psum[i-1] + temp; 
    }
    for (int i = K; i <= N; i++){
        ret = max(ret, psum[i] - psum[i - K]);
    }
    cout << ret << endl;
    return 0;
}

File: 백준/Silver/2559. 수열/수열.java
import java.io.BufferedReader;
import java.io.IOException;
import java.io.InputStreamReader;
import java.util.StringTokenizer;

public class Main {

    public static void main(String[] args) throws IOException {
        BufferedReader br = new BufferedReader(new InputStreamReader(System.in));
        StringTokenizer st = new StringTokenizer(br.readLine());

        int n = Integer.parseInt(st.nextToken());
        int k = Integer.parseInt(st.nextToken());
        int temp;
        int[] psum = new int[n + 1];
        int ret = -10000004;


        st = new StringTokenizer(br.readLine());
        for (int i = 1; i <= n; i++) {
            temp = Integer.parseInt(st.nextToken());
            psum[i] = psum[i - 1] + temp;
        }

        for (int i = k; i <= n; i++) {
            ret = Math.max(ret, psum[i] - psum[i - k]);
        }

        System.out.println(ret);
    }

}